### ISSUE TITLE
[Bug]: Table parsing abnormal

### ISSUE THREAD

**Issue by cyhasuka** [infiniflow/ragflow#2310]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

c18b78b

### Actual behavior

This bug appears in all versions of ragflow and is reproduced in the online demo.

1. Upload the simple test form below and parse it (using "Table" method). Note that I currently have and only have this one file in my knowledge base.
[test.xlsx](https://github.com/user-attachments/files/16924559/default.xlsx)
![image](https://github.com/user-attachments/assets/ee7ea9df-edf0-41b9-b0fe-28e550ab04ba)
![image](https://github.com/user-attachments/assets/9ad9ade3-522b-489f-95b6-29445f5e935e)

2. Ask anything about form filenames, the LLM output is as follows. 
![image](https://github.com/user-attachments/assets/f844f819-fc2d-4858-88df-df7ef631283d)

    Even if I delete the file from the knowledge base, the system still gives the above response when any word about the file name 
appears in the dialogue.
    Furthermore it return content in references that are not in this knowledge base, which can raise serious security issues.

### Steps to reproduce

```Markdown
As mentioned above.
```

**Comment by KevinHuSh** [comment]
'Table' is for content dumped from DB tables.
Have you tried 'General' to parse them?

**Comment by cyhasuka** [comment]
> 'Table' is for content dumped from DB tables.“表”用于从数据库表转储的内容。 Have you tried 'General' to parse them?您是否尝试过“常规”来解析它们？

Yes, it parses fine with ‘general’. If there is a problem with the way table resolves, should this be fixed or blocked?

**Comment by KevinHuSh** [comment]
It might not be suitable for 'Table'.

**Comment by ly0303521** [comment]
@KevinHuSh Hi, I'm parsing Excel file with method of "Table", and met the same problem, according to the manual I think Excel file is suitable for "Table". 
![Screenshot 2024-12-18 15:45:13](https://github.com/user-attachments/assets/a8bd282f-d4d2-4052-9cc0-37e923c28a10)
I use "Retrieval testing" to test, I found that the chunks is OK and retrieving is OK, so I think the Excel file is suitable for "Table".
Then I found the problem is caused by the code "use_sql" in "api/db/services/dialog_service.py", it will use the wrong references as you can see, finally I delete blow code, the method "Table" works well as I expected.
```
    # try to use sql if field mapping is good to go
    if field_map:
        chat_logger.info("Use SQL to retrieval:{}".format(questions[-1]))
        ans = use_sql(questions[-1], field_map, dialog.tenant_id, chat_mdl, prompt_config.get("quote", True))
        if ans:
            yield ans
            return
```
So this is a bug, right?

**Comment by cyhasuka** [comment]
> @KevinHuSh Hi, I'm parsing Excel file with method of "Table", and met the same problem, according to the manual I think Excel file is suitable for "Table".I use "Retrieval testing" to test, I found that the chunks is OK and retrieving is OK, so I think the Excel file is suitable for "Table". Then I found the problem is caused by the code "use_sql" in "api/db/services/dialog_service.py", it will use the wrong references as you can see, finally I delete blow code, the method "Table" works well as I expected.
> 
> ```
>     # try to use sql if field mapping is good to go
>     if field_map:
>         chat_logger.info("Use SQL to retrieval:{}".format(questions[-1]))
>         ans = use_sql(questions[-1], field_map, dialog.tenant_id, chat_mdl, prompt_config.get("quote", True))
>         if ans:
>             yield ans
>             return
> ```
> 
> So this is a bug, right?

I think you're right, this feature is useless and harmful to me, so I've blocked it as well.
